test-data/unit/check-modules.pyproject.test

```diff
@@ -0,0 +1,92 @@
+-- Type checker test cases dealing with modules and imports.
+-- Towards the end there are tests for PEP 420 (namespace packages, i.e. __init__.py-less packages).
+
+[case testModuleGetattrInit10PyprojectTOML]
+# flags: --config-file tmp/pyproject.toml
+import a.b.c  # silenced
+import a.b.d  # error
+
+[file a/__init__.pyi]
+from typing import Any
+def __getattr__(attr: str) -> Any: ...
+[file a/b/__init__.pyi]
+# empty (i.e. complete subpackage)
+
+[file pyproject.toml]
+\[tool.mypy]
+\[tool.'mypy-a.b.c']
```

Comment:
This looks a bit clunky to me. Maybe we can come up with a better way to represent this?

What about using the approach that was taken in the first attempt to implement pyproject.toml support (#5208)? It was like this:

```toml
[tool.mypy.overrides.'a.b.c']
ignore_missing_imports = true
```

This looks a bit cleaner to me, but still not great. 

I like this one more, as it allows splitting long lists of globs:

```toml
[[tool.mypy.overrides]]
module = "a.b.c"
ignore_missing_imports = true

[[tool.mypy.overrides]]
module = [
   "b.*",
   "c",
]
ignore_missing_imports = true
```

What do you think?

Reply:
My intention here was to keep it as close to the ini format as I could for ease of migration. I can certainly be swayed away from that. Either of the options shown look better. While I agree the second one looks cleaner, the downside to me is that now you're going to have to start checking things normally handled at the TOML level. For example, this would get flagged before it even gets parsed by our code in mypy:

```toml
[tool.mypy.overrides.'a.b.c']
ignore_missing_imports = true

[tool.mypy.overrides.'a.b.c']
ignore_missing_imports = false
```

However, under the second example, this is possible and would need code to account for it:

```toml
[[tool.mypy.overrides]]
module = "a.b.c"
ignore_missing_imports = true

[[tool.mypy.overrides]]
module = "a.b.c"
ignore_missing_imports = false
```

So now we'd have to start accounting for what happens when later overrides accidentally modify a prior override value. Do we keep oldest? Newest? Raise an error?

So out of laziness I would probably prefer the first as I'd like to lean on the TOML parser to catch situations like this.

It would take some work, but we could also support something like this, where the quotes are only needed for special characters:

```toml
[tool.mypy.overrides.a.b.c]
ignore_missing_imports = true

[tool.mypy.overrides.a.b.c.'*']
ignore_missing_imports = false
```

Let me know what you think @JukkaL

Reply:
At work we have sections with many globs, which I find pretty ugly. Here's an example with dummy names:

```
[mypy-pkg1.mod1,pkg2.subpkg.*,...<dozens of similar items>...,pkg30.something.else.*]
<various strictness options>
```

This makes me prefer the `[[tool.mypy.overrides]]` alternative. With this syntax, pyproject.toml would look more appealing than mypy.ini, in my opinion, and we could possibly start recommending it over mypy.ini in the future.

Note that the duplicate module name issue is possibly already a problem if we have multiple patterns per section:

```
[tool.mypy.overrides.'a.b,a.b.c']
...

[tool.mypy.overrides.'a.b.c']
...
```

Reply:
Okay, given that, what do we just say about duplicate behavior? Last one wins?

I'll start working on your second suggestion while we figure out this edge case.

Reply:
If there are two identical globs/module names, I think it's best to give an error. Otherwise it could be confusing, if the user doesn't realize that there are two sections that match.

Reply:
@JukkaL this should now be working the way you asked.